packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts

```diff
@@ -0,0 +1,298 @@
+  const ComboBoxRootBorderColor = palette.neutralTertiaryAlt;
+  const ComboBoxRootBorderColorHovered = semanticColors.inputFocusBorderAlt;
+  const ComboBoxRootColorErrored = semanticColors.errorText;
+  const ComboBoxCalloutBorderColor = palette.neutralLight;
+  const ComboBoxOptionHeaderTextColor = semanticColors.menuHeader;
+  const ComboBoxOptionDividerBorderColor = semanticColors.bodyDivider;
+
+  const styles: IComboBoxStyles = {
+    container: {
+
+    },
+    label: {
+
+    },
+    root: [
+      fonts.medium,
+      {
+        boxShadow: 'none',
+        margin: '0 0 10px 0',
+        padding: '0',
+        paddingRight: ComboxBoxCaretDownWidth,
+        color: ComboBoxRootTextColor,
+        position: 'relative',
+        outline: '0',
+        userSelect: 'none',
+        background: ComboBoxRootBackground,
+        borderWidth: '1px',
+        borderStyle: 'solid',
+        borderColor: ComboBoxRootBorderColor,
+        cursor: 'text',
+        display: 'block',
+        height: ComboBoxHeight,
+        lineHeight: ComboBoxLineHeight,
+        overflow: 'hidden',
+        whiteSpace: 'nowrap',
+        textOverflow: 'ellipsis',
+        boxSizing: 'content-box',
+        ' .ms-Label': {
+          display: 'inline-block',
+          marginBottom: '8px',
+        }
+      }
+    ],
+    rootHovered: {
+      borderColor: ComboBoxRootBorderColorHovered,
+      [MS_HIGHCONTRAST_ACTIVE]: {
+        color: 'HighlightText',
+        borderColor: 'Highlight'
+      },
+    },
+    rootFocused: {
+      borderColor: ComboBoxRootBorderColorHovered,
+      [MS_HIGHCONTRAST_ACTIVE]: {
+        color: 'HighlightText',
+        borderColor: 'Highlight'
+      },
+    },
+    rootDisabled: getDisabledStyles(theme),
+    rootError: {
+      borderColor: ComboBoxRootColorErrored
+    },
+    rootDisallowFreeForm: {
+
+    },
+    input: {
+      boxSizing: 'border-box',
+      width: '100%',
+      height: '100%',
+      border: 'none',
+      outline: 'none',
+      font: 'inherit',
+      textOverflow: 'ellipsis',
+      paddingLeft: '12px'
+    },
+    inputDisabled: getDisabledStyles(theme),
+    caretDownButtonStyles: null,
+    errorMessage: {
+      color: ComboBoxRootColorErrored,
+      ':before': {
+        content: '* ',
+      }
+    },
+    callout: {
+      boxShadow: '0 0px 5px 0px rgba(0, 0, 0, 0.4)',
+      borderWidth: '1px',
+      borderStyle: 'solid',
+      borderColor: ComboBoxCalloutBorderColor,
+    },
+    optionsContainer: {
+      display: 'block'
+    },
+    header: [
+      fonts.medium,
+      {
+        fontWeight: FontWeights.semibold,
+        color: ComboBoxOptionHeaderTextColor,
+        background: 'none',
+        border: 'none',
+        height: ComboBoxOptionHeight,
+        lineHeight: ComboBoxOptionHeight,
+        cursor: 'default',
+        padding: '0px 16px',
+        userSelect: 'none',
+        textAlign: 'left'
+      }
+    ],
+    divider: {
+      border: '1px solid',

```

Comment:
Suppose the customized style sets the following property:
borderColor: 'red' for the divider.

We would imagine that the computed border to be '1px solid red'. However, because the ordering of the object keys (the property names - border and borderColor here) is dependent on the JS runtime, we might have a situation where the borderColor: 'red' is overridden by the default style 

Checkout this codepen: https://codepen.io/hart/pen/QGJyKy?editors=0010